func.go

```diff
@@ -110,6 +110,9 @@ func funcRegular(L *lua.LState) int {
 			receiver = arg
 		} else {
 			arg = lValueToReflect(L, L.Get(i+1), hint, nil)
+			if !arg.IsValid() {
+				L.RaiseError("invalid type received for arg %d (expected %s)", i+1, hint)
```

Comment:
Unfortunately it'll be difficult to include the "got ..." part, unless we make significant modification to `lValueToReflect` to also return an "actual type". Do you have any preferences there? I think it's ok to omit the actual type - this error message is already a lot more helpful than what was returned previously.